imas/test/test_ids_convert.py

```diff
@@ -529,3 +529,46 @@ def test_3to4_migrate_deprecated_fields():  # GH#55
     del cp342.profiles_1d[0].ion[0].label
     cp4 = convert_ids(cp342, "4.0.0")
     assert cp4.profiles_1d[0].ion[0].name == "y"
+
+
+def test_3to4_name_identifier_mapping_cores_non_axisymmetric():
+    # Create source IDS using DD 3.39.0
+    pfa = IDSFactory("3.39.0")
+    if "cores_non_axisymmetric" not in pfa.ids_names():
+        pytest.skip("core_non_axisymmetric not present in DD 3.39.0 on this system")
+    src = pfa.cores_non_axisymmetric()
+    # Populate a parent that has name + identifier (no 'index' sibling)
+    src.coil.resize(1)
+    src.coil[0].name = "TEST_NAME"
+    src.coil[0].identifier = "TEST_IDENTIFIER"
+
+    # Convert to DD 4.0.0
+    dst = convert_ids(src, "4.0.0")
+
+    # Locate the destination node and assert mapping:
+    try:
+        dst4 = dst.core[0].name
+    except Exception:
+        pytest.skip("destination IDS structure does not contain core/name")
+
+    # DD3 name -> DD4 description
+    if hasattr(dst4, "description"):
+        assert dst4.description == "TEST_NAME"
+    else:
+        pytest.skip(
+            (
+                "DD4 destination does not have 'description' on this node - "
+                "cannot verify name->description mapping"
+            )
+        )
+
+    # DD3 identifier -> DD4 name
+    if hasattr(dst4, "name"):
+        assert dst4.name == "TEST_IDENTIFIER"
+    else:
+        pytest.skip(
+            (
+                "DD4 destination does not have 'name' on this node - "
+                "cannot verify identifier->name mapping"
+            )
+        )
```

Comment:
Elaborating on the previous comment: you don't need ` if hasattr(...)` because the IDS in DD4.0.0 is guaranteed to have these attributes:

```suggestion
    assert dst.coil[0].description == "TEST_NAME"
    assert dst.coil[0].name == "TEST_IDENTIFIER"
```